cpp/arcticdb/CMakeLists.txt

```diff
@@ -508,32 +506,18 @@ if (WIN32)
             /Zc:__cplusplus #needed to get the correct __cplusplus macro value
     )
 else()
-    if (STATIC_LINK_STD_LIB)
-        message("Statically linking standard libraries")
-        if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-             message(FATAL_ERROR "Statically linking standard libraries with CLANG is (currently) not supported")
-        elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-            set(standard_libraries
-                -static-libgcc
-                -static-libstdc++
-            )
-        else()
-            message(FATAL_ERROR "Unknown compiler: ${CMAKE_CXX_COMPILER_ID}")
-        endif()
+    message("Dynamically linking standard libraries")
+    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        set(standard_libraries
+            stdc++
+        )
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        set(standard_libraries
+            stdc++
+            gcc
+        )
```

Comment:
A few questions
1. Why do we have explicitly specify the standard libraries? I think CMake should be able to find them on it's own.
2. What does `gcc` do? I've never seen it.

Reply:
We need to figure this out (I only have dedented the code).

Reply:
- gcc provides a low-level runtime library, see [https://gcc.gnu.org/onlinedocs/gccint/Libgcc.html](https://gcc.gnu.org/onlinedocs/gccint/Libgcc.html)
- I think specifying these libraries was required in the case of static link (because by default cmake chooses the dynamic ones), and I guess it was added in the case of dynamic link for the sake of symmetry ;) This should be removed

Reply:
Yes, but:
 - Why do we need this low-level runtime library?
 - Do we need to stdc++ over libc++ for clang?

Reply:
[`manylinux2014` specifies `stdlibc++` but does not mention `libc++`](https://peps.python.org/pep-0599/#the-manylinux2014-policy); would it be the explanation?

Reply:
Resolved by f55ca51.